src/compute-engine/boxed-expression/match.ts

```diff
@@ -352,82 +403,217 @@ function matchArguments(
   }
 
   const ce = patterns[0].engine;
-  let result: BoxedSubstitution | null = { ...substitution };
 
   // We're going to consume the ops array, so make a copy
   const ops = [...expr.ops!];
 
-  let i = 0; // Index in pattern
-
-  while (i < patterns.length) {
-    const pat = patterns[i];
-    const argName = wildcardName(pat);
-
-    if (argName !== null) {
-      if (argName.startsWith('__')) {
-        // Match 1 or more expressions (__) or 0 or more (___)
-        /** Qty. of operands consumed (of those remaining to be matched). */
-        let j = 0;
-        if (patterns[i + 1] === undefined) {
-          // No more args in the pattern after, go till the end
-          j = ops.length;
-        } else {
-          // Capture til' the first matching arg. of the next pattern
-          let found = false;
-          const nextPat = patterns[i + 1];
-          while (!found && j < ops.length) {
-            found = matchOnce(ops[j], nextPat, result, options) !== null;
-            if (!found) j += 1;
-          }
-          // The next pattern does not match against any of the remaining ops.. Unless the next
-          // match is optional, then can assume no overall match.
-          if (!found && !wildcardName(nextPat)?.startsWith('___')) return null;
-        }
-
-        // Determine the value to return for the wildcard
-        let value: BoxedExpression;
-        if (j < 1) {
-          // A standard/non-optional Sequence Wildcard with no match...
-          if (!argName.startsWith('___') && j < 1) return null;
-          // Otherwise must be an optional-sequence match: this is permitted.
-          if (expr.operator === 'Add') value = ce.Zero;
-          else if (expr.operator === 'Multiply') value = ce.One;
-          else value = ce.Nothing;
-        } else if (j === 1) {
-          // Capturing a single element/operand
-          value = ops.shift()!;
-        } else {
-          // >1 operands captured
-          const def = ce.lookupDefinition(expr.operator);
-          const args = ops.splice(0, j);
-          if (def && isOperatorDef(def) && def.operator.associative) {
-            value = ce.function(expr.operator, args, { canonical: false });
+  return matchRemaining(patterns, substitution);
```

Comment:
Recursive-calling necessary due to requirement of attempting variations of operand-capture qty. for sequences (see commit message).